src/hist.jl

```diff
@@ -60,6 +60,44 @@ push!(Plots._histogram_like, :ea_histogram)
     ()
 end
 @shorthands testhist
+
+# ---------------------------------------------------------------------------
+# grouped histogram
+
+@userplot GroupedHist
+
+@recipe function f(p::GroupedHist)
+    @assert length(p.args) == 2
+    v = p.args[1]
+    gs = p.args[2]
+
+    bins = get(plotattributes, :bins, :auto)
+    normed = get(plotattributes, :normalize, false)
+    weights = get(plotattributes, :weights, nothing)
+
+    # compute edges from ungrouped data
+    h = Plots._make_hist((v,), bins; normed = normed, weights = weights)
+
+    # compute weights (frequencies) by group using those edges
+    nbins = length(h.weights)
+    groups = sort(unique(gs))
```

Comment:
It may be better to use `extractGroupArgs` as in [GroupedBoxplot](https://github.com/JuliaPlots/StatsPlots.jl/blob/master/src/boxplot.jl#L149). The reason is that it implements a few extra things (i.e. better legend if one groups by named tuples) that would be lost if one does a custom grouping implementation here.